src/AspNet.Security.OAuth.HubSpot/HubSpotAuthenticationHandler.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.HubSpot
+{
+    public class HubSpotAuthenticationHandler : OAuthHandler<HubSpotAuthenticationOptions>
+    {
+        public HubSpotAuthenticationHandler(
+            [NotNull] IOptionsMonitor<HubSpotAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint + "/" + tokens.AccessToken);
```

Comment:
Is this correct? Seems unusual to have the access token sent as part of the path, rather than in a header or the query string.

```suggestion
            using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint + "/" + tokens.AccessToken);
```

Reply:
Sadly, that is how HubSpot did it. Their APIs are far from uniform but they're working on it. Also, it might surprise you that their tokens are just GUIDs.